gtsfm/data_association/data_assoc.py

```diff
@@ -68,21 +72,27 @@ def __validate_track(self, sfm_track: Optional[SfmTrack]) -> bool:
         """Validate the track by checking its length."""
         return sfm_track is not None and sfm_track.numberMeasurements() >= self.min_track_len
 
-    def run(
+    def run_da(
         self,
         num_images: int,
         cameras: Dict[int, gtsfm_types.CAMERA_TYPE],
         tracks_2d: List[SfmTrack2d],
+        sfm_tracks: List[Optional[SfmTrack]],
+        avg_track_reproj_errors: List[Optional[float]],
+        triangulation_exit_codes: List[TriangulationExitCode],
         cameras_gt: List[Optional[gtsfm_types.CALIBRATION_TYPE]],
         relative_pose_priors: Dict[Tuple[int, int], Optional[PosePrior]],
         images: Optional[List[Image]] = None,
     ) -> Tuple[GtsfmData, GtsfmMetricsGroup]:
-        """Perform the data association.
+        """Perform the data association and compute metrics.
```

Comment:
Since we don't triangulate here, and don't compute correspondences here, I propose we totally rename this method to `compute_da_metrics()`.

Reply:
(and edit the docstring also to remove the mention of "Perform the data association", since all we do is compute metrics here know if I understand correctly)

Reply:
Re-reading this, looks like we do also validate the tracks here, and compute a largest connected component. How about we consider renaming this as
```python
def validate_tracks_and_compute_metrics()
```
or
```python
def validate_triangulated_tracks_and_compute_metrics()
  """ """
```
or 
```python
def assemble_gtsfm_data_from_tracks()
```
How about this as a docstring:
  """Assembles GtsfmData, validates triangulated tracks, computes metrics.

Only the largest connected component of cameras, represented in 3d tracks, is retained.
"""

Reply:
I think these are good suggestions, but again I think it makes more sense to save this for another PR in which you can refactor the module names and functions to your liking.